pkg/cloudproviders/aws/aws.go

```diff
@@ -232,30 +241,35 @@ func NewProviderFromViper(v *viper.Viper, logger logrus.FieldLogger, _ string) (
 	if err := http2.ConfigureTransport(transport); err != nil {
 		return nil, err
 	}
-	sharedConfig := aws.NewConfig().
-		WithHTTPClient(&http.Client{
+
+	cfg, err := config.LoadDefaultConfig(context.TODO(),
+		config.WithHTTPClient(&http.Client{
 			Transport: transport,
 			Timeout:   httpTimeout,
-		}).
-		WithMaxRetries(a.GetInt("max_retries"))
-	metadataSession, err := session.NewSession(sharedConfig)
-	if err != nil {
-		return nil, fmt.Errorf("error creating a new Metadata session: %v", err)
-	}
-	metadata := ec2metadata.New(metadataSession)
-	region, err := metadata.Region()
+		}),
+		config.WithRetryMaxAttempts(5),
```

Comment:
Should we be setting that from max_retries config?

Reply:
Yes we should, I thought I had updated that, weird